core/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.action.admin.indices.rollover;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request class to swap index under an alias given some predicates
+ * TODO: documentation
+ */
+public class RolloverRequest extends AcknowledgedRequest<RolloverRequest> implements IndicesRequest {
+
+    private String sourceAlias;
+    private String optionalTargetAlias;
+    private List<Condition> conditions = new ArrayList<>();
+
+    RolloverRequest() {}
+
+    public RolloverRequest(String sourceAlias, String optionalTargetAlias) {
+        this.sourceAlias = sourceAlias;
+        this.optionalTargetAlias = optionalTargetAlias;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (sourceAlias == null) {
+            validationException = addValidationError("source alias is missing", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        sourceAlias = in.readString();
+        optionalTargetAlias = in.readOptionalString();
+        int size = in.readVInt();
```

Comment:
I think NamedWriteable gives you the flexibility to do more with conditions in a backwards compatible way later. So, yeah, I think NamedWriteable is better than just Writeable.